include/umf/providers/provider_level_zero.h

```diff
@@ -68,6 +68,21 @@ umf_result_t umfLevelZeroMemoryProviderParamsSetResidentDevices(
     umf_level_zero_memory_provider_params_handle_t hParams,
     ze_device_handle_t *hDevices, uint32_t deviceCount);
 
+typedef enum umf_level_zero_memory_provider_free_policy_t {
+    UMF_LEVEL_ZERO_MEMORY_PROVIDER_FREE_POLICY_DEFAULT =
```

Comment:
Can we somehow test different policies?

Reply:
That's a bit problematic. We would need to have a test that launches a kernel, and that means an extra dependency on a compiler or committing a binary. As a first step, I wanted to test this in UR (that's why I kept this as draft). Once we have  test there, I will think if we can easily port it to this repo.

Reply:
Sounds good. I asked just in case there is a simple test. In future, we can consider running integration tests with UR/compiler.